remote-config/examples/remote_config_fetch.rs

```diff
@@ -0,0 +1,94 @@
+const SERVICE: &str = "testservice";
+const ENV: &str = "testenv";
+const VERSION: &str = "1.2.3";
+
+#[tokio::main(flavor = "current_thread")]
+async fn main() {
+    // SingleChangesFetcher is ideal for a single static (runtime_id, service, env, version) tuple
+    // Otherwise a SharedFetcher (or even a MultiTargetFetcher for a potentially high number of
+    // targets) for multiple targets is needed. These can be manually wired together with a
+    // ChangeTracker to keep track of changes. The SingleChangesTracker does it for you.
+    let mut fetcher = SingleChangesFetcher::new(
+        // Use SimpleFileStorage if you desire just the raw, unparsed contents
+        // (e.g. to do processing directly in your language)
+        // For more complicated use cases, like needing to store data in shared memory, a custom
+        // FileStorage implementation is recommended
+        ParsedFileStorage::default(),
+        Target {
+            service: SERVICE.to_string(),
+            env: ENV.to_string(),
+            app_version: VERSION.to_string(),
+        },
+        RUNTIME_ID.to_string(),
+        ConfigInvariants {
+            language: "awesomelang".to_string(),
+            tracer_version: "99.10.5".to_string(),
+            endpoint: Endpoint {
+                url: hyper::Uri::from_static("http://localhost:8126"),
+                api_key: None,
+            },
+            products: vec![ApmTracing],
+            capabilities: vec![],
+        },
+    );
+
+    // Custom timeout, defaults to 5 seconds.
+    fetcher.set_timeout(2000);
+
+    loop {
+        match fetcher.fetch_changes().await {
+            Ok(changes) => {
+                println!("Got {} changes:", changes.len());
+                for change in changes {
+                    match change {
+                        Change::Add(file) => {
+                            println!("Added file: {} (version: {})", file.path(), file.version());
+                            print_file_contents(&file.contents());
+                        }
+                        Change::Update(file, _) => {
+                            println!(
+                                "Got update for file: {} (version: {})",
+                                file.path(),
+                                file.version()
+                            );
+                            print_file_contents(&file.contents());
+                        }
+                        Change::Remove(file) => {
+                            println!("Removing file {}", file.path());
+                        }
+                    }
+                }
+            }
+            Err(e) => {
+                eprintln!("Fetch failed with {e}");
+                fetcher.set_last_error(e.to_string());
+            }
+        }
+
+        sleep(Duration::from_nanos(fetcher.get_interval()).max(Duration::from_secs(1))).await;
+    }
+}
+
+fn print_file_contents(contents: &anyhow::Result<RemoteConfigData>) {
```

Comment:
file contents might be multi megabytes, you should at most print a few bytes of it, this code is a dangerous pattern.

Reply:
This is an example, to debug and inspect RC contents. In actual code you'd process it and not print it.

Reply:
it's going to be taken as an example

Reply:
I'll add a warning in a comment about the possible size.